src/pkg/quota/aws_quotas.go

```diff
@@ -0,0 +1,115 @@
+package quota
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/servicequotas"
+	"github.com/compose-spec/compose-go/v2/types"
+)
+
+var (
+	gpuQuotaCodes = []string{"L-7212CCBC", "L-3819A6DF"} // these are the GPU quota codes from cd
+	serviceCode   = "ec2"
+)
+
+type QuotaClientAPI interface {
+	ListServiceQuotas(ctx context.Context, params *servicequotas.ListServiceQuotasInput, optFns ...func(*servicequotas.Options)) (*servicequotas.ListServiceQuotasOutput, error)
+}
+
+type AwsServiceQuotas struct {
+	quotaClient QuotaClientAPI
+	Gpus        float32
+}
+
+func NewAwsServiceQuotas(ctx context.Context, cfg aws.Config) *AwsServiceQuotas {
+	return &AwsServiceQuotas{
+		quotaClient: servicequotas.NewFromConfig(cfg),
+		Gpus:        -1,
+	}
+}
+
+func (q *AwsServiceQuotas) Initialize(ctx context.Context, cfg aws.Config) error {
+	accountQuota, err := q.GetAccountGpuQuota(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting account quota: %v", err)
+	}
+
+	if len(accountQuota) == 0 {
+		return errors.New("no quotas returned")
+	}
+
+	cpuCount := make([]float32, 0, len(accountQuota))
+	for _, v := range accountQuota {
+		cpuCount = append(cpuCount, v)
+	}
+
+	// sort descending
+	sort.Slice(cpuCount, func(i, j int) bool { return cpuCount[i] > cpuCount[j] })
+	q.Gpus = float32(cpuCount[0]) // get the largest GPU quota across all instances
```

Comment:
CPU count != GPU count. The smallest GPU instance (1 GPU) has 4 CPUs, if I remember correctly.

Reply:
It's also totally plausible to use all G and all P instances simultaneously, so why only pick the largest?